openslides_backend/action/actions/speaker/create.py

```diff
@@ -56,21 +61,56 @@ def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
             instance["weight"] = weight_max + 1
             return instance
 
-        list_of_speakers_id = instance["list_of_speakers_id"]
-        weight_no_poos_min = self._get_no_poo_min(
-            list_of_speakers_id, instance["meeting_id"]
+        meeting = self.datastore.get(
+            fqid_from_collection_and_id("meeting", instance["meeting_id"]),
+            ["point_of_order_category_enabled"],
         )
-        if weight_no_poos_min is None:
-            instance["weight"] = weight_max + 1
-            return instance
+        list_of_speakers_id = instance["list_of_speakers_id"]
+        if meeting.get("point_of_order_category_enabled"):
+            filter = And(
+                FilterOperator("list_of_speakers_id", "=", list_of_speakers_id),
+                FilterOperator("begin_time", "=", None),
+                FilterOperator("meeting_id", "=", instance["meeting_id"]),
+            )
+            speakers = self.datastore.filter(
+                self.model.collection,
+                filter=filter,
+                mapped_fields=[
+                    "id",
+                    "weight",
+                    "point_of_order",
+                    "point_of_order_category_id",
+                ],
+            )
+            los = sorted(speakers.values(), key=lambda k: k["weight"])
+            index = len(los) - 1
+            new_speaker_rank = self.get_rank(instance["point_of_order_category_id"])
```

Comment:
Read all the point_of_order_categories with their rank once before the loop and use it here.